autogpt_platform/frontend/src/components/OttoChatWidget.tsx

```diff
@@ -0,0 +1,340 @@
+}
+
+interface Message {
+  type: "user" | "assistant";
+  content: string;
+}
+
+const OttoChatWidget = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [inputValue, setInputValue] = useState("");
+  const [isProcessing, setIsProcessing] = useState(false);
+  const [includeGraphData, setIncludeGraphData] = useState(false);
+  const messagesEndRef = useRef<HTMLDivElement>(null);
+  const { user, supabase } = useSupabase();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const flowID = searchParams.get("flowID");
+  const { nodes, edges } = useAgentGraph(
+    flowID ? (flowID as GraphID) : undefined,
+  );
+  const { toast } = useToast();
+
+  useEffect(() => {
+    // Add welcome message when component mounts
+    if (messages.length === 0) {
+      setMessages([
+        {
+          type: "assistant",
+          content: "Hello im Otto! Ask me anything about AutoGPT!",
+        },
+      ]);
+    }
+  }, [messages.length]);
+
+  useEffect(() => {
+    // Scroll to bottom whenever messages change
+    messagesEndRef.current?.scrollIntoView({ behavior: "smooth" });
+  }, [messages]);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!inputValue.trim() || isProcessing) return;
+
+    const userMessage = inputValue.trim();
+    setInputValue("");
+    setIsProcessing(true);
+
+    // Add user message to chat
+    setMessages((prev) => [...prev, { type: "user", content: userMessage }]);
+
+    try {
+      if (!supabase) {
+        throw new Error("Supabase client not initialized");
+      }
+
+      // Get the current session
+      const {
+        data: { session },
+      } = await supabase.auth.getSession();
+      if (!session) {
+        throw new Error("No active session");
+      }
+
+      const messageId = `${Date.now()}-web`;
+
+      const payload = {
+        query: userMessage,
+        conversation_history: messages.reduce<
+          { query: string; response: string }[]
+        >((acc, msg, i, arr) => {
+          if (
+            msg.type === "user" &&
+            i + 1 < arr.length &&
+            arr[i + 1].type === "assistant"
+          ) {
+            acc.push({
+              query: msg.content,
+              response: arr[i + 1].content,
+            });
+          }
+          return acc;
+        }, []),
+        user_id: user?.id || "anonymous",
+        message_id: messageId,
+        include_graph_data: includeGraphData,
+        graph_id: flowID || undefined,
+      };
+
+      setIncludeGraphData(false);
+
+      // Add temporary processing message
+      setMessages((prev) => [
+        ...prev,
+        { type: "assistant", content: "Processing your question..." },
+      ]);
+
+      const BACKEND_URL =
+        process.env.NEXT_PUBLIC_BACKEND_URL || "http://localhost:8006";
+      const response = await fetch(`${BACKEND_URL}/api/otto/ask`, {
```

Comment:
Do all this in a server action

Reply:
Also move this call to the client api, but also still call that client in a server actions